public/src/main/java/com/revenuecat/purchases/models/GoogleStoreProduct.kt

```diff
@@ -99,6 +126,36 @@ data class GoogleStoreProduct(
     )
     override val sku: String
         get() = productId
+
+    /**
+     * For internal RevenueCat use.
+     *
+     * Creates a copy of this `GoogleStoreProduct` with the specified `offeringId` set on itself and its
+     * `defaultOption`/`subscriptionOptions`.
+     */
+    override fun copyWithOfferingId(offeringId: String): StoreProduct {
+        val subscriptionOptionsWithOfferingIds = subscriptionOptions?.mapNotNull {
+            (it as? GoogleSubscriptionOption)?.let { googleOption ->
+                GoogleSubscriptionOption(googleOption, offeringId)
+            }
+        }
+
+        val defaultOptionWithOfferingId = (defaultOption as? GoogleSubscriptionOption)?.let {
+            GoogleSubscriptionOption(
+                it,
+                offeringId
+            )
+        }
+
+        return GoogleStoreProduct(
+            this,
+            defaultOptionWithOfferingId,
+            if (subscriptionOptionsWithOfferingIds.isNullOrEmpty()) null else SubscriptionOptions(
+                subscriptionOptionsWithOfferingIds
+            ),
```

Comment:
Should this be replaced with a `subscriptionOptionsWithOfferingIds?.map` so that the empty list is preserved if it was one? 

Right now this will change the `subscriptionOptions` to null if it was originally an empty list 🤔